lib/engine/tile.rb

```diff
@@ -206,7 +206,7 @@ def terrain
       @upgrades.flat_map(&:terrains).uniq
     end
 
-    def upgrades_to?(other, special = false)
+    def upgrades_to?(game, other, special = false)
```

Comment:
i don't think we should make this change or ghave label_check_override, 

didn't the other games just add the label or remove it specially?

Reply:
I have not come up with a better solution. Further comment below.

Reply:
ok i've thought about it a bit. move upgrades_to? out of tile and into game, this will require shifting around some specs but i think it makes sense it will give the game full control over upgrade rules

Reply:
Ok, I will implement that change. I think it can be applied to 18AL as well - the special handling for Montgomery can then be done in its upgrades_to?

Using that (in 18AL) I suppose it does not need to add any labels. Instead the upgrades_to? will make the green M tile available when upgrading to Green, and exclude the non M green tiles.

And the Wiki page should be updated for 18AL (and 18TN) to reflect this.

I will probably do this implementation later today. I intend to create a spec file first, so that it will manage regression test for 18TN when doing this change. Should be possible to add these commits. Do not think there is any conflicts in master.
(Assuming there is not problem with other spec files - we will see.)